events/wc_offmap_events.txt

```diff
@@ -707,33 +877,389 @@ narrative_event = {
+				limit = { has_wonder = wonder_ahnqiraj }
+				save_event_target_as = target_wonder
+			}
+		}
+		else_if = {
+			limit = { nzoth_offmap_decisions_enabled_trigger = yes }
+			random_realm_wonder = {
+				limit = { has_wonder = wonder_shrine_storm }
+				save_event_target_as = target_wonder
+			}
+		}
+		else = {
+			random_realm_wonder = {
+				limit = { has_wonder = wonder_ulduar }
+				save_event_target_as = target_wonder
+			}
+		}
+	}
+	
+	option = {
+		name = EVTOPTA_WCOME.300
+		if = {
+			limit = { cthun_offmap_decisions_enabled_trigger = yes }
+			if = {
+				limit = { offmap_cthun = { has_status = old_god_imprisoned } }
+				offmap_cthun = { set_status = old_god_partly_imprisoned }
+				clr_character_flag = freeing_old_god_flag
+			}
+			else_if = {
+				limit = { offmap_cthun = { has_status = old_god_partly_imprisoned } }
+				offmap_cthun = { set_status = old_god_partly_free }
+				event_target:target_wonder = {
+					add_upgrade_slots = 1
+					add_wonder_upgrade = upgrade_old_god_cthun_1
+				}
+				narrative_event = { id = WCOME.360 }
+			}
+			else_if = {
+				limit = { offmap_cthun = { has_status = old_god_partly_free } }
+				offmap_cthun = { set_status = old_god_free }
+				event_target:target_wonder = {
+					destroy_wonder_upgrade = upgrade_old_god_cthun_1
+					add_wonder_upgrade = upgrade_old_god_cthun_2
+				}
+				clr_character_flag = freeing_old_god_flag
+			}
+		}
+		else_if = {
+			limit = { nzoth_offmap_decisions_enabled_trigger = yes }
+			if = {
+				limit = { offmap_nzoth = { has_status = old_god_imprisoned } }
+				offmap_nzoth = { set_status = old_god_partly_imprisoned }
+				clr_character_flag = freeing_old_god_flag
+			}
+			else_if = {
+				limit = { offmap_nzoth = { has_status = old_god_partly_imprisoned } }
+				offmap_nzoth = { set_status = old_god_partly_free }
+				event_target:target_wonder = {
+					add_upgrade_slots = 1
+					add_wonder_upgrade = upgrade_old_god_nzoth_1
+				}
+				narrative_event = { id = WCOME.360 }
+			}
+			else_if = {
+				limit = { offmap_nzoth = { has_status = old_god_partly_free } }
+				offmap_nzoth = { set_status = old_god_free }
+				event_target:target_wonder = {
+					destroy_wonder_upgrade = upgrade_old_god_nzoth_1
+					add_wonder_upgrade = upgrade_old_god_nzoth_2
+				}
+				clr_character_flag = freeing_old_god_flag
+			}
+		}
+		else_if = {
+			limit = { yogg_saron_offmap_decisions_enabled_trigger = yes }
+			if = {
+				limit = { offmap_yoggsaron = { has_status = old_god_imprisoned } }
+				offmap_yoggsaron = { set_status = old_god_partly_imprisoned }
+				d_ulduar = {
+					holder_scope = {
+						any_realm_character = {
+							religion = old_gods_worship
+							set_government_type = black_pact_government
+						}
+						religion = old_gods_worship
+						set_government_type = black_pact_government
+						any_realm_province = {
+							remove_province_modifier = ulduar_closed_down
+							religion = old_gods_worship
+						}
+					}
+				}
```

Comment:
Ulduar isn't closed since 11f7a2d69a4849789b81d1139d83567718b2a670.
Moreover, it needs a redesign. It should be openable from outside.